src/node_url.cc

```diff
@@ -1145,9 +1147,23 @@ namespace url {
           }
           break;
         case kPathStart:
-          state = kPath;
-          if (ch != '/' && !special_back_slash)
-            continue;
+          if (IsSpecial(url->scheme)) {
+            state = kPath;
+            if (ch != '/' && ch != '\\') {
+              continue;
+            }
+          } else if (!has_state_override && ch == '?') {
+            url->flags |= URL_FLAGS_HAS_QUERY;
```

Comment:
Sorry for not being clear. Keep the `url->query.clear()` there, in addition to setting the flag.